Mage/src/main/java/mage/watchers/Watcher.java

```diff
@@ -114,96 +112,7 @@ public <T extends Watcher> T copy() {
             for (Field field : allFields) {
                 if (!Modifier.isStatic(field.getModifiers())) {
                     field.setAccessible(true);
-
-                    if (field.getType() == Set.class) {
-                        // Set<UUID, xxx>
-                        ((Set) field.get(watcher)).clear();
-                        ((Set) field.get(watcher)).addAll((Set) field.get(this));
-                    } else if (field.getType() == Map.class || field.getType() == HashMap.class) {
-                        // Map<UUID, xxx>
-                        ParameterizedType parameterizedType = (ParameterizedType) field.getGenericType();
-                        Type valueType = parameterizedType.getActualTypeArguments()[1];
-                        if (valueType.getTypeName().contains("SortedSet")) {
-                            // Map<UUID, SortedSet<Object>>
```

Comment:
Check new code for SortedSet copy - it must keep it after copy.

Reply:
I check for a TreeSet directly